toolkit/docs/building/add-package.md

```diff
@@ -0,0 +1,404 @@
+mkdir -p $TARBALL_FOLDER
+cp $SRC_TARBALL $tmpdir
+
+pushd $tmpdir > /dev/null
+
+PKG_NAME="ig"
+NAME_VER="$PKG_NAME-$PKG_VERSION"
+VENDOR_TARBALL="$OUT_FOLDER/$NAME_VER-govendor-v$VENDOR_VERSION.tar.gz"
+
+echo "Unpacking source tarball..."
+tar -xf $SRC_TARBALL
+
+echo "Vendor go modules..."
+cd inspektor-gadget-"$PKG_VERSION"
+go mod vendor
+
+echo ""
+echo "========================="
+echo "Tar vendored tarball"
+tar  --sort=name \
+     --mtime="2021-04-26 00:00Z" \
+     --owner=0 --group=0 --numeric-owner \
+     --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime \
+     -cf "$VENDOR_TARBALL" vendor
+
+popd > /dev/null
+echo "$PKG_NAME vendored modules are available at $VENDOR_TARBALL"
+```
+
+We now need to call the script:
+
+```bash
+$ bash generate_source_tarball.sh --srcTarball ig-0.25.0.tar.gz --pkgVersion 0.25.0
+...
+$ ls
+generate_source_tarball.sh  ig-0.25.0-govendor-v1.tar.gz  ig.spec  ig-0.25.0.tar.gz
+```
+
+## Building the package
+
+We have all the files needed to build the package, let's build it:
+
+```bash
+$ cd toolkit
+# First, we need to prepare to get the toolchain:
+$ ./scripts/setuplkgtoolchain.sh
+...
+Finished syncing toolchain to LKG ('3-0-20240227' - '7d886f3a947fb53ff3c08fd95823f1b0398e3d2f')
+To download LKG toolchain, run:
+        sudo make toolchain -j8 REBUILD_TOOLCHAIN=n REBUILD_TOOLS=y DAILY_BUILD_ID=3-0-20240227
+# Let's get the toolchain:
+$ sudo make toolchain -j8 REBUILD_TOOLCHAIN=n REBUILD_TOOLS=y DAILY_BUILD_ID=3-0-20240227
+...
+Downloading toolchain RPM: ...
+# Let's build our package.
+# NOTE, if you also changed the packaged used as BuildRequires, you should add in all *_LIST.
+$ sudo make build-packages REBUILD_TOOLS=y SOURCE_URL='https://cblmarinerstorage.blob.core.windows.net/sources/core' PACKAGE_BUILD_LIST='ig' PACKAGE_REBUILD_LIST='ig' SRPM_PACK_LIST='ig' RUN_CHECK=y SRPM_FILE_SIGNATURE_HANDLING=update DAILY_BUILD_ID=3-0-2024028
+...
+INFO[0003][srpmpacker] Packing 1/1 SPECs
+...
+INFO[0000][scheduler] Testing: ig-0.25.0-1.azl3.src.rpm
+INFO[0000][scheduler] Building: ig-0.25.0-1.azl3.src.rpm
+...
+INFO[0200][scheduler] Built SRPMs:
+INFO[0200][scheduler] --> ig-0.25.0-1.azl3.src.rpm
+INFO[0200][scheduler] Passed SRPMs tests:
+INFO[0200][scheduler] --> ig-0.25.0-1.azl3.src.rpm
+```
+
+Everything was built and tested, you should now have the package available:
+
+```bash
+$ rpm -ql ../out/RPMS/x86_64/ig-0.25.0-1.azl3.x86_64.rpm
+/usr/bin/ig
+/usr/share/licenses/ig
+/usr/share/licenses/ig/LICENSE
+/usr/share/licenses/ig/LICENSE-bpf.txt
+```
+
+The content seems good, let's test it locally:
+
+```bash
+$ rpm2cpio ../out/RPMS/x86_64/ig-0.25.0-1.azl3.x86_64.rpm | cpio -idmv
+./usr/bin/ig
+./usr/share/licenses/ig
+./usr/share/licenses/ig/LICENSE
+./usr/share/licenses/ig/LICENSE-bpf.txt
+149181 blocs
+$ ./usr/bin/ig --help
+Collection of gadgets for containers
+
+Usage:
+  ig [command]
+...
+```
+
+Congratulations! You successfully built a CBL-Mariner package!
+
```

Comment:
We have a few automatic tools in the toolkit to help here, setting up a full VM can be a pain.

Feel free to edit the formatting, the suggestion box didn't like using backticks.
```suggestion
## Testing in a Container

The toolkit can generate an azl containers for you, and auto launch them:
```bash
# Start the test container
sudo make containerized-rpmbuild MODE=test SRPM_PACK_LIST='ig' DAILY_BUILD_ID=3-0-2024028
...
~/repos/temp/CBL-Mariner_TEMP3/toolkit/scripts/containerized-build/create_container_build.sh 
...
Populating Intermediate SRPMs...
...
Setting up tools...
Setting up mounts...
Importing chroot into docker...
Chroot is up-to-date
Checking if build env is up-to-date...
...
 -----------------------------------------------------------------------------------------
 ----------------------------------- MARINER BUILDER ! ----------------------------------- 
 -----------------------------------------------------------------------------------------
******************************************************************************************
* To see this menu again, run  show_help
*
* Some tips:
*     Use local RPMs to satify dependencies:        run enable_local_repo
...

# Create a rpm repo based on the contents of ./out/RPMS/
enable_local_repo
...
-------- enabling local repo ---------
...
# Install the new package
tdnf install /mnt/RPMS/x86_64/ig-0.25.0-1.azl3.x86_64.rpm
ig --help
```